src/SonarQube.Client/Services/SonarQubeClient.cs

```diff
@@ -188,40 +216,84 @@ public Task<Result<PropertyResponse[]>> GetPropertiesAsync(CancellationToken tok
 
         public Task<Result<VersionResponse>> GetVersionAsync(CancellationToken token)
         {
-            const string ServerVersionAPI = "api/server/version"; // Since 2.10; internal
+            // Since 2.10; internal
+            var serverVersionAPI = new Uri("api/server/version", UriKind.Relative);
 
-            return InvokeSonarQubeApi(ServerVersionAPI, token,
+            return InvokeSonarQubeApi(serverVersionAPI, token,
                 stringResponse => new VersionResponse { Version = stringResponse });
         }
 
         public Task<Result<CredentialResponse>> ValidateCredentialsAsync(CancellationToken token)
         {
-            const string ValidateCredentialsAPI = "api/authentication/validate"; // Since 3.3
+            // Since 3.3
+            var validateCredentialsApiUrl = new Uri("api/authentication/validate", UriKind.Relative);
 
-            return InvokeSonarQubeApi(ValidateCredentialsAPI, token,
+            return InvokeSonarQubeApi(validateCredentialsApiUrl, token,
                 stringResponse => new CredentialResponse { IsValid = (bool)JObject.Parse(stringResponse).SelectToken("valid") });
         }
 
-        private static string AppendQueryString(string urlBase, string queryFormat, params object[] args)
+        public Task<Result<NotificationsResponse[]>> GetNotificationEventsAsync(NotificationsRequest request,
+            CancellationToken token)
         {
-            return urlBase +
-                string.Format(CultureInfo.InvariantCulture, queryFormat,
-                    args.Select(x => HttpUtility.UrlEncode(x.ToString())).ToArray());
+            // Since 6.6; internal
+            var apiPath = BuildRelativeUrl("api/developers/search_events",
+                new Dictionary<string, string>
+                {
+                    ["projects"] = request.ProjectKey,
+                    ["from"] = ToJavaTimeFormat(request.EventsSince)
+                });
+
+            return InvokeSonarQubeApi(apiPath, token,
+                    async (HttpResponseMessage httpResponse) =>
+                    {
+                        if (httpResponse.StatusCode != HttpStatusCode.OK)
+                        {
+                            return new NotificationsResponse[0];
+                        }
+
+                        var stringResponse = await GetStringResultAsync(httpResponse, token);
+                        return JObject.Parse(stringResponse)["events"].ToObject<NotificationsResponse[]>()
+                            ?? new NotificationsResponse[0];
+                    }
+                );
+        }
+
+        private static string ToJavaTimeFormat(DateTimeOffset date)
```

Comment:
I would extract this method into some helper class and add more intensive testing.